scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+def get_repository_commit_sha(repository_location):
+    command = "git -C " + repository_location + " rev-parse HEAD"
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
+    else:
+        return subprocess.CompletedProcess(process.args, process.returncode, stdout, stderr).stdout.decode('ascii').strip()
+
+def update_ncs(repository_location, version):
+    # Checkout sdk-nrf to the desired version.
+    command = "git -C " + repository_location + " checkout " + version
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
+
+    # Call west update command to update all projects and submodules used by sdk-nrf.
+    command = "west update -r"
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
+
+def get_ncs_recommended_version():
+    chip_root = os.path.dirname(os.path.abspath(__file__)).replace("scripts/setup/nrfconnect", "")
+
+    # Read recommended version saved in the .nrfconnect-recommended-version file.
+    try:
+        with open(os.path.join(chip_root, 'config/nrfconnect/.nrfconnect-recommended-version'), 'r') as f:
+            return f.readline().strip()
+    except:
+        raise RuntimeError("Encountered problem when trying to read .nrfconnect-recommended-version file.")
+
+def print_check_version_warning_message(current_version, recommended_version):
+    current_version_message = "WARNING: Your current NCS version (" + current_version + ")"
```

Comment:
I guess you need this: https://docs.python.org/3/library/textwrap.html#textwrap.wrap :)